lib/Doctrine/ORM/Tools/Pagination/Paginator.php

```diff
@@ -163,7 +163,7 @@ public function getIterator()
             $ids          = array_map('current', $foundIdRows);
 
             $this->appendTreeWalker($whereInQuery, WhereInWalker::class);
-            $whereInQuery->setHint(WhereInWalker::HINT_PAGINATOR_ID_COUNT, count($ids));
+            $whereInQuery->setHint(WhereInWalker::HINT_PAGINATOR_ID_COUNT, (bool) $ids);
```

Comment:
I'd either do this:
```suggestion
            $whereInQuery->setHint(WhereInWalker::HINT_PAGINATOR_ID_COUNT,  count($ids) > 0);
```
or this:
```suggestion
            $whereInQuery->setHint(WhereInWalker::HINT_PAGINATOR_ID_COUNT,  $ids !== []);
```

Relying on a cast from array to bool feels unsafe and hard to grasp, it might be why we are generally trying to avoid this in new code.

Reply:
Good suggestion, changed that :+1:.